salt/_pillar/velum.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+'''
+A module that adds data to the Pillar structure retrieved by an http request made to Velum
+
+
+Configuring the VELUM ext_pillar
+====================================
+
+Set the following Salt config to setup Foreman as external pillar source:
+
+.. code-block:: json
+
+  ext_pillar:
+    - velum:
+        - url: https://velum.example.com/internal-api/v1/pillar?version=1
+        - ca_bundle: /some/bundle.crt
+        - username: username
+        - password: secret
+
+Module Documentation
+====================
+'''
+
+# Import python libs
+from __future__ import absolute_import
+import logging
+
+# Import Salt libs
+import salt.ext.six as six
+
+import json
+import hashlib
+
+pillar_sha256 = None
+pillar_cache = None
+cached_username = None
+cached_password = None
```

Comment:
Why do we need these global variables, is that because they persist across the whole life cycle of a salt-master instance?

Reply:
Yes, their purpose is clear it me. I just wanted to double check about their "persistence".